### PR TITLE
Editorial for Moving to the Capital

### PR DESCRIPTION
**Pull request by eysbutno**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: solutions/gold/cf-1472G.mdx [hunk 1]
```diff
@@ -0,0 +1,107 @@
+---
+id: cf-1472G
+source: Codeforces
+title: Moving to the Capital
+author: Justin Ji
+---
+
+<Spoiler title = "Hint #1">
```
Comment (SansPapyrus683) [inline solutions/gold/cf-1472G.mdx:8]: just "Hint 1" instead of "Hint 2" and the same for the others

File: solutions/gold/cf-1472G.mdx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+void solve() {
+	int n, m;
+	cin >> n >> m;
+	vector<vector<int>> adj(n);
+	for (int i = 0; i < m; i++) {
+		int u, v;
+		cin >> u >> v;
+		--u, --v;
+		adj[u].push_back(v);
```
Comment (SansPapyrus683) [inline solutions/gold/cf-1472G.mdx:61]:
```suggestion
		u--, v--;
		adj[u].push_back(v);
```

File: solutions/gold/cf-1472G.mdx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+				min_dist[node_1] = min(min_dist[node_1], capital_dist[node_2]);
+			}
+		}
+	}
+	for (int i = 0; i < n; i++) { cout << min_dist[i] << " \n"[i == n - 1]; }
+}
+int main() {
+	int t;
+	cin >> t;
+	while (t--) { solve(); }
```
Comment (SansPapyrus683) [inline solutions/gold/cf-1472G.mdx:100]:
```suggestion
	int test_num;
	cin >> test_num;
	for (int t = 0; t < test_num; t++) { solve(); }
```
and just put the body of `solve()` right inside here while you're at it

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
so true bestie

**Review by envyaims — APPROVED** [review]
fire

File: solutions/gold/cf-1472G.mdx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+		// order the nodes by distance from capital
+		vector<int> order(n);
+		iota(begin(order), end(order), 0);
+		sort(begin(order), end(order),
+		     [&](int i, int j) { return capital_dist[i] > capital_dist[j]; });
+
+		// calculate the DP
+		vector<int> min_dist = capital_dist;
+		for (int node_1 : order) {
+			for (int node_2 : adj[node_1]) {
```
Comment (SansPapyrus683) [inline solutions/gold/cf-1472G.mdx:90]: rename to `node1` and `node2`

File: solutions/gold/cf-1472G.mdx [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
+	int test_num;
+	cin >> test_num;
+	for (int tt = 0; tt < test_num; tt++) {
```
Comment (SansPapyrus683) [inline solutions/gold/cf-1472G.mdx:56]:
```suggestion
	for (int tc = 0; tc < test_num; tc++) {
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
almost done i swear

File: solutions/gold/cf-1472G.mdx (hunk 4 above)
Comment (TheGamingMousse) [reply solutions/gold/cf-1472G.mdx]: done

File: solutions/gold/cf-1472G.mdx (hunk 5 above)
Comment (TheGamingMousse) [reply solutions/gold/cf-1472G.mdx]: done

**Review by SansPapyrus683 — APPROVED** [review]

**Pull request merged**